websites/G/GitHub/presence.ts

```diff
@@ -1,294 +1,228 @@
-
-  pullRequestTitle = issueTitle = document.querySelector(
-    "div.gh-header-show h1 span.js-issue-title"
-  );
-  pullRequestAuthor = issueAuthor = document.querySelectorAll(
-    "div div.timeline-comment-header.clearfix h3 strong a"
-  );
-  pullRequestID = issueID = document.querySelector(
-    "#span.f1-light.text-gray-light"
-  );
-
-  let profileTabs: string, profileCurrentTab: string;
-
-  if (profileName) {
-    profileTabs = "/" + profileNickname.innerText + "?tab=";
-    profileCurrentTab = profileURL.searchParams.get("tab");
+  const profile: { [key: string]: HTMLElement } = {
+      name: document.querySelector(".vcard-names .p-name"),
+      nickname: document.querySelector(".vcard-names .p-nickname")
+    },
+    organization: { [key: string]: HTMLElement } = {
+      name: document.querySelector(
+        "#js-pjax-container > div > header > div.container-lg.pt-4.pt-lg-0.p-responsive.clearfix > div > div.flex-auto.ml-3.ml-md-4 > h1"
+      )
+    },
+    repository: { [key: string]: HTMLElement } = {
+      author: document.querySelector(".author a"),
+      name: document.querySelector(
+        "#js-repo-pjax-container > div > div > div > h1 > strong > a"
+      ),
+      location: document.querySelector(
+        "#branch-select-menu > summary > span.css-truncate-target"
+      )
+    },
+    action: { [key: string]: HTMLElement } = {
+      title: document.querySelector(
+        "#repo-content-pjax-container > div > div.js-updatable-content.js-socket-channel.d-flex.flex-items-start.flex-md-items-center.pb-3.pb-md-4.pl-0.pl-md-2.mt-n2.mb-1.ml-1 > div.d-flex.flex-auto.mr-3 > h3 > span"
+      )
+    },
+    PRandIssues: { [key: string]: HTMLElement } = {
+      title: document.querySelector(
+        "div.gh-header-show h1 span.js-issue-title"
+      ),
+      id: document.querySelector("#span.f1-light")
+    },
+    nodeListOf: { [key: string]: NodeListOf<HTMLElement> } = {
+      PRandIssueAuthor: document.querySelectorAll(
+        "div div.timeline-comment-header.clearfix h3 strong a"
+      ),
+      repLoc2: document.querySelectorAll("#blob-path")
+    },
+    browsingStamp = Math.floor(Date.now() / 1000),
+    searchURL = new URL(document.location.href),
+    searchResult = searchURL.searchParams.get("q");
+
+  let profileTabs: string,
+    profileCurrentTab: string;
+
+  if (profile.name) {
+    profileTabs = "/" + profile.nickname.innerText + "?tab=";
+    profileCurrentTab = new URL(document.location.href).searchParams.get("tab");
   }
 
   const presenceData: PresenceData = {
-    details: "Unknown page",
-    largeImageKey: "lg"
+    details: "Unsupported page",
+    largeImageKey: "lg",
+    startTimestamp: browsingStamp
   };
 
   if (document.location.pathname == "/" || !document.location.pathname) {
-    presenceData.state = "Home";
-
-    presenceData.startTimestamp = browsingStamp;
-
-    delete presenceData.details;
+    presenceData.details = "Home";
+    delete presenceData.state;
```

Comment:
I meant removing state completly from the code, since now you are deleting state when its not set.
```suggestion
    presenceData.details = "Home";
```